hazelcast/include/hazelcast/util/SynchronizedQueue.h

```diff
@@ -50,6 +51,21 @@ namespace hazelcast {
                 return e;
             }
 
+            size_t size() {
+                util::LockGuard lg(m);
+                return internalQueue.size();
+            }
+
+            std::vector<boost::shared_ptr<T> > values() {
+                util::LockGuard lg(m);
+                std::vector<boost::shared_ptr<T> > values;
+                for (typename std::deque<boost::shared_ptr<T> >::const_iterator it = internalQueue.begin();
```

Comment:
If we had something like:

```
for (auto &it: internalQueue){
values.push_back(it);
}
```

Would it have the behavior?

Reply:
We use BOOST_FOREACH but I can not put it here at this header not to cause external dependency sşnce this header may be accessed via user included headers. But we use BOOST_FOREACH internally.